### PR TITLE
add suffix option to avoid hardcoded addition of /*.log

### PR DESCRIPTION
**Pull request by jp-paillet**
Hi all,
This proposal is for fixing the issue "Input docker behaviour with Openshift 3.10 #8988"
Thanks.

### CODE REVIEW

**Comment by ruflin** [comment]
@jp-paillet Thank you for the PR. To get CI running on this, could you run `make fmt` an push again?

Could you also add a changelog entry?

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -6,6 +6,96 @@
...
+
+==== Added
+
+*Affecting all Beats*
+
+*Auditbeat*
+
+*Filebeat*
+
+- Added `containers.suffix` input configuration option to avoid the static addition of the "/*.log" in paths. {issue}8988[8988] {pull}9620[9620]
```
Comment (ruflin) [inline CHANGELOG.asciidoc]: You can just add this under `Beats version HEAD` into the right section. No need to introduce all this content. Should be in the end only be 1 line.

Comment (jp-paillet) [reply CHANGELOG.asciidoc]: Hi Nicolas, thanks for your clarifications, but I'm now lost into git actions... I'll try to end this by today anyway.

Comment (jp-paillet) [reply CHANGELOG.asciidoc]: Hi Nicolas, I wish my last changes will feet your expectations. I'll be on holidays for the next 2 weeks to come, so I'll look at this in January. Again, thanks for your guidance. Merry Christmas.

Comment (ruflin) [reply CHANGELOG.asciidoc]: @jp-paillet Thanks for pushing this forward. We will probably not get to review this one today but lets try to get this in beginning next year. Merry Christmas, happy new year 🎉

File: Jenkinsfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-library identifier: 'apm@master',
+library identifier: 'apm@v1.0.2',
```
Comment (jsoriano) [inline Jenkinsfile]: Changes in this file don't seem related.

File: filebeat/input/docker/input.go [hunk 3]
```diff
@@ -68,8 +69,8 @@ func NewInput(
 		return nil, errors.New("Docker input requires at least one entry under 'containers.ids'")
 	}
 
-	for idx, containerID := range ids {
-		cfg.SetString("paths", idx, path.Join(config.Containers.Path, containerID, "*.log"))
+	for idx, containerID := range config.Containers.IDs {
```
Comment (jsoriano) [inline filebeat/input/docker/input.go]: Is there a reason to don't use `ids`? It filters empty docker ids.

File: filebeat/input/docker/input.go [hunk 4]
```diff
@@ -68,8 +69,8 @@ func NewInput(
...
+		cfg.SetString("paths", idx, path.Join(config.Containers.Path, strings.Join([]string{containerID, config.Containers.Suffix}, "")))
```
Comment (jsoriano) [inline filebeat/input/docker/input.go]: With default suffix `*.log` (without the `/`) this could also be:
```suggestion
		cfg.SetString("paths", idx, path.Join(config.Containers.Path, containerID, config.Containers.Suffix))
```

Comment (jsoriano) [reply filebeat/input/docker/input.go]: Or this is intended to support any suffix?

File: filebeat/input/docker/config.go [hunk 5]
```diff
@@ -35,13 +36,14 @@ type config struct {
...
+	// Force CRI format (don't perform autodetection)
 	CRIForce bool `config:"cri.force"`
 }
 
 type containers struct {
-	IDs  []string `config:"ids"`
-	Path string   `config:"path"`
+	IDs    []string `config:"ids"`
+	Path   string   `config:"path"`
+	Suffix string   `config:"suffix"`
```
Comment (jsoriano) [inline filebeat/input/docker/config.go]: Another option could be to accept glob-based paths here as in the [log input](https://www.elastic.co/guide/en/beats/filebeat/6.5/filebeat-input-log.html#input-paths), this way the default could be `/var/lib/docker/containers/<container_id>/*.log`, and the one for cri-o `/var/log/containers/*-*.log.
I see the combination of path and suffix a bit confussing.

**Comment by ruflin** [comment]
@jp-paillet For the changelog: We just changed a few days ago how we handle the changelog and introduced CHANGELOG.next.asciidoc. Perhaps you can move your entry there and reset the CHANGELOG.asciidoc to what is in master?

File: Jenkinsfile (hunk 2 above)
Comment (jp-paillet) [reply Jenkinsfile]: Hi Jaime, well honestly I don't known how this came in, but for sure it seems useless for me too. I'll come back to "master" in my next push then.

File: filebeat/input/docker/input.go (hunk 3 above)
Comment (jp-paillet) [reply filebeat/input/docker/input.go]: Well, here, again, I just used the code that was here when I created my repo. I'll see if using ids is fine.
Thanks.

**Comment by jp-paillet** [comment]
I'm sorry, but I'm not able to finish this properly : I'm lost inside git intricacies, and my knowledge of the go language is too limited. The current content of my branch is missing most of the initial code, so I think it can be send to trash.
The issue #8988 is no more a problem with us : My initial patch has been implemented on our servers and is running well.
So, I think this PR can be closed.
Thanks.
